### PR TITLE
Add ZenCode tab, add applyFlight in ArmorSetEffects, start work on Preprocessors and advanced ZC things

### PR DESCRIPTION
**Pull request by Witixin1512**
I also moved the Arrays file there, let me know if you don't like that

### CODE REVIEW

File: docs/1.16/content/zencode/ConditionalCode.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Conditional Code
+
+Onto more complex scripting! Conditional code is an important aspect of intermediate to advanced scripts.
+
+Conditional code entails running a chunk of code if a `boolean` condition evaluates to `true`.
+This allows you to execute code that can adapt to conditions in the world, the player, a block, or the entity that's dropping items.
+
+# If Statement
+
+An if statement executed the piece of code if the `boolean` inside of its bracket evaluates to `true`.
```
Comment (jaredlll08) [inline docs/1.16/content/zencode/ConditionalCode.md:10]:
```suggestion
An if statement executes the piece of code if the `boolean` inside of its bracket evaluates to `true`.
```

File: docs/1.16/content/zencode/Preprocessors/Preprocessors.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Preprocessors
+
+Preprocessors are, as their name suggests, a list of markers a script can have applied to it that tag the script with information when it is compiled.
+They run various actions on scripts. A preprocessor is called by appending `#preprocessorname [arguments]` to the top of your script. Be careful not to accidentally enable one through the use of a comment!
```
Comment (jaredlll08) [inline docs/1.16/content/zencode/Preprocessors/Preprocessors.md:4]: pre procs aren't always at the top, they can actually be anywhere, like the `#snip` preproc.

**Review by jaredlll08 — CHANGES_REQUESTED** [review]

File: docs/1.16/content/zencode/Preprocessors/Preprocessors.md (hunk 2 above)
Comment (Witixin1512) [reply docs/1.16/content/zencode/Preprocessors/Preprocessors.md:4]: I actually forgot about the `#snip` preprocessor, that's gonna be annoying

Comment (jaredlll08) [reply docs/1.16/content/zencode/Preprocessors/Preprocessors.md:4]: There are actually quite a few:
```
#debug
#loader
#loadfirst
#loadlast
#modloaded
#noload
#priority
#replace
```

and snips:
```
#snip modloaded
#snip modnotloaded
#snip side
#snip start
#snip end
```

Comment (Witixin1512) [reply docs/1.16/content/zencode/Preprocessors/Preprocessors.md:4]: I thought I included a list with doc links with some of them in the prepocessors file...
I'm gonna double check that soon

Comment (Witixin1512) [reply docs/1.16/content/zencode/Preprocessors/Preprocessors.md]: Fixed that, added more preprocessors to the file, I'll look into what they do exactly. Was looking into this yesterday, I might ask you about some things (https://github.com/CraftTweaker/CraftTweaker/tree/1.16/src/main/java/com/blamejared/crafttweaker/api/zencode/impl/preprocessors)

I presume `noload` does what the name claims, `debug` generates classes, `loadfirst` and `loadlast` are direct changes to priority +-1? Or do they literally make the scripts load first or last depending on other's scripts priority? What does `replace` do? I _think_ I know what `snip` does, its a fragment of a script that only loads on X or Y side and if the conditions match. What if you want to combine params? Also, do you need to put the params next to the `start` call or are they on another line?) 

Just realised this is a lot of text so if you'd rather answer in discord go ahead lol
Thanks!

**Review by jaredlll08 — APPROVED** [review]

**Pull request merged**